frontend/src/App.js

```diff
@@ -74,81 +84,187 @@ function App() {
+                color="inherit"
+                style={{ textTransform: "none" }}
+                onClick={handleRandomize}
+              >
+                <h3>Random</h3>
+              </Button>
+            </Grid>
+          </Grid>
+        </Toolbar>
+      </AppBar>
+      <div style={{ padding: "1rem" }}>
         {matrices?.map((matrix, matrixIndex) => (
           <div key={matrixIndex}>
             <h2>
               Enter a {rows}x{cols} Matrix ({matrixIndex + 1})
             </h2>
             {matrix.map((row, rowIndex) => (
               <div key={rowIndex}>
-                {row.map((col, colIndex) => (
-                  <input
-                    key={`${matrixIndex}-${rowIndex}-${colIndex}`}
-                    type="text"
-                    value={col}
-                    onChange={(e) =>
-                      handleChange(
-                        matrixIndex,
-                        rowIndex,
-                        colIndex,
-                        e.target.value
-                      )
-                    }
-                    style={{ width: "50px", marginRight: "10px" }}
-                  />
-                ))}
-                <br />
+                <Grid container spacing={1}>
+                  {row.map((col, colIndex) => (
+                    <Grid item xs>
+                      {total ? (
+                        <TextField
+                          key={`${matrixIndex}-${rowIndex}-${colIndex}`}
+                          type="number"
+                          value={col}
+                          onChange={(e) =>
+                            handleChange(
+                              matrixIndex,
+                              rowIndex,
+                              colIndex,
+                              e.target.value
+                            )
+                          }
+                          style={{
+                            margin: "10px",
+                          }}
+                        />
+                      ) : (
+                        <input
+                          key={`${matrixIndex}-${rowIndex}-${colIndex}`}
+                          type="number"
+                          value={col}
+                          onChange={(e) =>
+                            handleChange(
+                              matrixIndex,
+                              rowIndex,
+                              colIndex,
+                              e.target.value
+                            )
+                          }
+                          style={{
+                            width: "30px",
+                          }}
+                        />
+                      )}
+                    </Grid>
+                  ))}
+                  <br />
+                </Grid>
               </div>
             ))}
           </div>
         ))}
-        <button type="submit">Send Matrices</button>
-        {submitted && <h3> Matrices Sent!</h3>}
```

Comment:
I think we should still have feedback that the matrices have been sent, either add this back, the alert back, or some other form of feedback. Super important for user experience if calc is slow for w/e reason and for debugging during a live demo, as feedback is faster than checking network tab.

Reply:
Would prefer some sort of non-intrusive alert, so no popup pls. I found it very annoying because it would always steal focus and had to be dismissed manually.

Reply:
Wait nvm I'm thinking of the "failed to send matrices" popup